### PR TITLE
BD-3677: Add troubleshooting for few or no direct opens for iOS push

### PR DESCRIPTION
**Pull request by bre-fitzgerald**
### Why are you making this change? (required)

Resolves BD-3677 as part of the [initiative](https://confluence.braze.com/display/PDT/KB+article+tracker) to source docs feedback from KB articles.

### Related PRs, issues, or features (optional)

- [BD-3677](https://jira.braze.com/browse/BD-3677)

### Feature release date (optional)

- N/A

### Contributor checklist

- [x] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 1]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
 
+_Direct Opens_ are not logged for test sends or silent push notifications. If you have sent an iOS push notification and know at least one user has clicked the notification, but few or no _Direct Opens_ have been logged, this is likely an integration issue. 
+
+- Make sure that the messages are not being sent as [silent push notifications][1]. The message must have text in the title or body to not be considered silent.
+- Check that you have set up push notifications according to the integration guide. Any form of click reporting, including _Direct Opens_, requires the following steps to be implemented correctly:
+   - [Step 3:][2] On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
+   - [Step 5:][3] Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```
Comment (bre-fitzgerald) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:170]:
```suggestion
   - [Step 3][2]: On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
   - [Step 5][3]: Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 2]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
 
 If opens are being logged, check whether it is an issue with the deep link in general or with the deep linking push click handling. To do this, test to see if a deep link from an in-app message click works.
 
+#### Few or no direct opens
 
+_Direct Opens_ are not logged for test sends or silent push notifications. If you have sent an iOS push notification and know at least one user has clicked the notification, but few or no _Direct Opens_ have been logged, this is likely an integration issue.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:165]:
```suggestion
If at least one user opens your iOS push notification, but few or no _Direct Opens_ are logged to Braze, there may be an issue with your [SDK integration]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/initial_sdk_setup/overview). Keep in mind, _Direct Opens_ are not logged for test sends or silent push notifications.
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 3]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
 
 If opens are being logged, check whether it is an issue with the deep link in general or with the deep linking push click handling. To do this, test to see if a deep link from an in-app message click works.
 
+#### Few or no direct opens
 
+_Direct Opens_ are not logged for test sends or silent push notifications. If you have sent an iOS push notification and know at least one user has clicked the notification, but few or no _Direct Opens_ have been logged, this is likely an integration issue. 
+
+- Make sure that the messages are not being sent as [silent push notifications][1]. The message must have text in the title or body to not be considered silent.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:167]:
```suggestion
- Messages sent without title or body text are sent as [silent push notifications][1]. Verify that your messages have  both title and body text.
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 4]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
+- Check that you have set up push notifications according to the integration guide. Any form of click reporting, including _Direct Opens_, requires the following steps to be implemented correctly:
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:168]:
```suggestion
- Double-check the following steps from the [push integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration):
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 5]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
+   - [Step 3][2]: On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:169]:
```suggestion
   - **[Register for push notifications][{{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration/#step-3-register-for-push-notifications]:** On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 6]
```diff
@@ -160,4 +160,15 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
+   - [Step 5][3]: Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:170]:
```suggestion
   - [Enable push handling][{{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration/#step-5-enable-push-handling]: Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```

**Review by internetisaiah — CHANGES_REQUESTED** [review]
No need to use my suggestions 1 for 1, but I cleaned up the content of the KB article to be more clear.

**Comment by bre-fitzgerald** [comment]
Thanks @internetisaiah! Great edits, accepted 🫡

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 7]
```diff
@@ -160,4 +160,12 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
+- Double-check the following steps from the [push integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration):
+   - [Step 3]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-1-register-for-push-notifications-with-apns): On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
+   - [Step 5]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration/#step-5-enable-push-handling): Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:170]:
```suggestion
   - [Enable push handling]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration/#step-5-enable-push-handling): Verify that the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method has been implemented.
```

File: _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md [hunk 8]
```diff
@@ -160,4 +160,12 @@ Most of the code that handles deep links also handles push opens. First, ensure
...
+
+- Make sure that the messages are not being sent as [silent push notifications]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/silent_push_notifications/#sending-silent-push-notifications). The message must have text in the title or body to not be considered silent.
+- Double-check the following steps from the [push integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/integration):
+   - [Step 3]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-1-register-for-push-notifications-with-apns): On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/legacy_sdks/ios/push_notifications/troubleshooting.md:169]:
```suggestion
   - [Register for push]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-1-register-for-push-notifications-with-apns): On every single app launch, preferably within `application:didFinishLaunchingWithOptions:`, the code from step 3 needs to occur. The delegate property of `UNUserNotificationCenter.current()` needs to be assigned to an object that implements `UNUserNotificationCenterDelegate` and contains the `(void)userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` method.
```

**Review by internetisaiah — APPROVED** [review]
LGTM!

**Pull request merged**